test/boost/sstable_compaction_test.cc

```diff
@@ -1217,6 +1218,96 @@ SEASTAR_TEST_CASE(tombstone_purge_test) {
     });
 }
 
+SEASTAR_TEST_CASE(mv_tombstone_purge_test) {
+    BOOST_REQUIRE(smp::count == 1);
+    return test_env::do_with_async([] (test_env& env) {
+        // In a column family with gc_grace_seconds set to 0, check that a tombstone
+        // is purged after compaction.
+        auto builder = schema_builder("tests", "tombstone_purge")
+                .with_column("id", utf8_type, column_kind::partition_key)
+                .with_column("ck", int32_type, column_kind::clustering_key)
+                .with_column("value", int32_type);
+        builder.set_gc_grace_seconds(0);
+        auto s = builder.build();
+
+        auto sst_gen = env.make_sst_factory(s);
+
+        auto compact = [&, s] (std::vector<shared_sstable> all, std::vector<shared_sstable> to_compact) -> std::vector<shared_sstable> {
+            auto cf = env.make_table_for_tests(s);
+            auto stop_cf = deferred_stop(cf);
+            for (auto&& sst : all) {
+                column_family_test(cf).add_sstable(sst).get();
+            }
+            return compact_sstables(env, sstables::compaction_descriptor(to_compact), cf, sst_gen).get().new_sstables;
+        };
+
+        auto next_timestamp = [] {
+            static thread_local api::timestamp_type next = 1;
+            return next++;
+        };
+
+        auto make_insert = [&] (partition_key key, int32_t ck = 0, int32_t value = 1, std::optional<api::timestamp_type> timestamp = std::nullopt, std::optional<api::timestamp_type> created_at = std::nullopt) {
+            mutation m(s, key);
+            if (!timestamp) {
+                created_at = timestamp = next_timestamp();
+            } else if (!created_at) {
+                created_at = next_timestamp();
+            }
+            auto c_key = clustering_key::from_single_value(*s, int32_type->decompose(data_value(ck)));
+            testlog.info("make_insert: key={} ck={} timestamp={} created_at={}", dht::decorate_key(*s, key), ck, *timestamp, *created_at);
+            m.set_clustered_cell(
+                    c_key,
+                    bytes("value"),
+                    data_value(value),
+                    *timestamp);
+            m.partition().clustered_row(*s, c_key).apply(row_marker(*created_at));
+            return m;
+        };
+
+        auto make_delete_row = [&] (partition_key key, int32_t ck = 0, gc_clock::time_point deletion_time = gc_clock::now(), std::optional<api::timestamp_type> timestamp = std::nullopt, std::optional<api::timestamp_type> deleted_at = std::nullopt) {
```

Comment:
Does the `timestamp` parameter serve any role in this function? I don't see it being read anywhere in the function.

Reply:
You're right.
It was used in an early implementation of this function but its usage got dropped.
I'll drop the parameter then

Reply:
Will be dropped in next version